### PR TITLE
Check xfopen

### PR DESCRIPTION
**Pull request by r4f4**

### CODE REVIEW

File: fus.c [hunk 1]
```diff
@@ -394,6 +401,11 @@ filelist_loadcb (Pool     *pool,
   fpath = repodata_lookup_str (data, SOLVID_META, REPOSITORY_REPOMD_LOCATION);
 
   fp = solv_xfopen (fpath, 0);
+  if (!fp)
+    {
+      g_info ("Could not open filelists %s: %s", fpath, g_strerror (errno));
+      return 0;
```
Comment (lubomir) [inline fus.c:413]: All other places return `NULL`.

**Review by lubomir — COMMENTED** [review]
I think the messages should be printed as error. If the repo can not be found, that is serious problem and even if the process returns some results, they will quite likely not be correct. For the same reason it could make sense to not just ignore incorrect repos, but just exit the process unsuccessfully.

File: fus.c (hunk 1 above)
Comment (r4f4) [reply fus.c:413]: I can't return NULL here because this is a callback and libsolv expects either a 0 or 1 to know if the loading was successfull or not.

**Comment by r4f4** [comment]
Ok, I can change it so that an invalid repo will make the process exit with an error.
But you mean just not finding a repomd.xml should trigger the error, right? Maybe if there is not primary metadata inside it too? What about group/group_gz?

File: fus.c (hunk 1 above)
Comment (lubomir) [reply fus.c:413]: Ah, that makes sense.

**Review by lubomir — APPROVED** [review]
Good point. Missing group/group_gz is definitely no problem. It definitely needs to treat different files differently, so I guess the current approach is good.

**Comment by r4f4** [comment]
I change it so that an error is returned in case any of the repos passed as input cannot be created. I am still not sure what to do in case there is a repomd.xml but it doesn't contain valid/any data. In its current state, fus would fail to find solvables and/or to resolve the dependencies.

**Comment by lubomir** [comment]
I think what you have now is good. It catches the most common issue that the path to repo is wrong or it can't be opened. I wouldn't worry about repos that miss primary.xml or it contains garbage. Groups, modules, filelists etc. can be missing and it's not a big deal, it's still a valid repo.

File: fus.c [hunk 2]
```diff
@@ -326,11 +327,20 @@ static Repo *
...
                   const char *name,
                   const char *type,
-                  const char *path)
+                  const char *path,
+                  GError    **error)
 {
+  FILE *fp = fopen (path, "r");
+  if (!fp)
+    {
+      g_autofree char *msg = g_strdup_printf("Could not open %s: %s",
```
Comment (igor-raits) [inline fus.c]: `func ()` style please

File: fus.c [hunk 3]
```diff
@@ -326,11 +327,20 @@ static Repo *
...
+                                             path, g_strerror (errno));
+      g_set_error_literal (error, G_OPTION_ERROR, G_OPTION_ERROR_FAILED, msg);
```
Comment (igor-raits) [inline fus.c]: why not to use g_set_error and not create `msg` variable?

File: fus.c [hunk 4]
```diff
@@ -403,25 +420,36 @@ filelist_loadcb (Pool     *pool,
...
   const char *fname;
 
-  Repo *repo = repo_create (pool, name);
-
   fp = solv_xfopen (pool_tmpjoin (pool, path, "/", "repodata/repomd.xml"), "r");
+  if (!fp)
+    {
+      g_autofree char *msg = g_strdup_printf("Could not open repomd.xml for %s: %s",
+                                             path, g_strerror (errno));
+      g_set_error_literal (error, G_OPTION_ERROR, G_OPTION_ERROR_FAILED, msg);
```
Comment (igor-raits) [inline fus.c]: same here

**Comment by r4f4** [comment]
Thank you for the review. I submitted updates addressing the raised issues.

File: fus.c [hunk 5]
```diff
@@ -326,11 +327,21 @@ static Repo *
 create_test_repo (Pool       *pool,
                   const char *name,
                   const char *type,
-                  const char *path)
+                  const char *path,
+                  GError    **error)
```
Comment (igor-raits) [inline fus.c]: Add one more space and reindent others please

File: fus.c [hunk 6]
```diff
@@ -403,25 +421,37 @@ filelist_loadcb (Pool     *pool,
 static Repo *
 create_repo (Pool       *pool,
              const char *name,
-             const char *path)
+             const char *path,
+             GError    **error)
```
Comment (igor-raits) [inline fus.c]: Same

**Review by igor-raits — COMMENTED** [review]
Apart from few style nitpicks looks good. Merge it yourself once they are fixed.

**Comment by r4f4** [comment]
I don't have permission to merge PRs.

**Pull request merged**